boa/src/exec/mod.rs

```diff
@@ -343,9 +327,9 @@ impl Interpreter {
             ValueData::Object(_) => {
                 let prim_value = self.to_primitive(&mut (value.clone()), Some("number"));
                 self.to_string(&prim_value)
-                    .to_string()
+                    .expect("string")
```

Comment:
Could we add a more explanatory message here?

Reply:
I would have liked to use the `?` operator, but the `to_number` is not spec compliant when I make `to_number` spec compliant i will use `?`.